src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorSpanHelpers.cs

```diff
@@ -222,7 +222,7 @@ public static void ValidateStrides(ReadOnlySpan<nint> strides, ReadOnlySpan<nint
         /// <param name="length">The length of the TensorSpan we are iterating over.</param>
         public static void AdjustIndexes(int curIndex, nint addend, Span<nint> curIndexes, scoped ReadOnlySpan<nint> length)
         {
-            if (addend <= 0 || curIndex < 0)
+            if (addend <= 0 || curIndex < 0 || length[curIndex] <= 0)
```

Comment:
What's hitting this? Will this affect behavior of other public API?

Reply:
It removes Access violation when length[curIndex] is zero. Now code throws an Access violation for an empty tensor.
That was incorrect behavior. I add tests for empty tensor.

Reply:
@michaelgsharp